testlapack/dlaqr1.go

```diff
@@ -0,0 +1,91 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package testlapack
+
+import (
+	"math"
+	"math/rand"
+	"testing"
+
+	"github.com/gonum/floats"
+)
+
+type Dlaqr1er interface {
+	Dlaqr1(n int, h []float64, ldh int, sr1, si1, sr2, si2 float64, v []float64)
+}
+
+func Dlaqr1Test(t *testing.T, impl Dlaqr1er) {
+	rnd := rand.New(rand.NewSource(1))
+
+	for _, n := range []int{2, 3} {
```

Comment:
I just wanted to have a single loop body for all the four possible cases. If you want, I can factor the body out into `dlaqr1Test(t *testing.T, impl Dlaqr1er, n, cas int)` and call it here 4 times with fixed arguments.